cloudevents/working-drafts/dds-protocol-binding.md

```diff
@@ -0,0 +1,265 @@
+content mode MUST support the [JSON event format][json-format].
+
+The DDS protocol binding additionally supports the
+[DDS event format][dds-message-format].
+
+### 1.5. Security
+
+This specification does not introduce any new security features for DDS, or
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+### 2.1. data
+
+`data` is assumed to contain opaque application data that is encoded as declared
+by the `datacontenttype` attribute. 
+
+There are two values of `datacontenttype` currently supported:
+- `cloudevent/json` for JSON data
+- `application/cloudevent+dds` for binary and plain text data types
+
+If the `datacontenttype` attribute is set to `application/cloudevent+dds`, the
+`datacontentencoding` attribute defines the encoding of the message body. 
+
+The `content-type` field in the message header MUST be consistent with the
+`datacontenttype` attribute.
+
+## 3. DDS Message Mapping
+
+The content type is chosen by the sender of the event. The receiver of the
+event can distinguish between content modes by inspecting the `content-type`
+header of the DDS message.
+
+If the header is present and its value is `cloudevent/json`, the receiver
+decodes the message as JSON data.
+
+If the header is present and its value is `application/cloudevent+dds`, the
+receiver decodes the message into the DDS message format, using the
+`datacontentencoding` attribute to determine the encoding of the message body.
+There are three valid values of the `datacontentencoding` attribute
+currently supported:
+- `text` for ASCII text
+- `binary` for binary data
+
+The `content-type` of the DDS message is REQUIRED to be consistent with the
+`datacontenttype` attribute.
+
+If a receiver finds a CloudEvents type as per the above rule, but with a
+`datacontentencoding` that it cannot handle, it MAY treat the event as binary
+and forward it to another party as-is.
+
+When the `content-type` header value is not set, knowing when the message ought
+to be parsed as a CloudEvent can be a challenge. While this specification
+can not mandate that senders do not include any of the CloudEvents headers
+when the message is not a CloudEvent, it would be reasonable for a receiver
+to assume that if the message has all of the mandatory CloudEvents attributes
+as headers then it`s probably a CloudEvent. However, as with all CloudEvent
+messages, if it does not adhere to all of the normative language of this
+specification then it is not a valid CloudEvent.
+
+### 3.1. Keys
+
+The `datakey` of the DDS message MAY be populated. A _key field_ in DDS is way
+to uniquely identify individual instances of data being published to a topic. For
+example, if you are publishing data to a Temperature Topic dealing with
+temperature readings from different sensors, the sensor ID could be a key field.
```

Comment:
This example feels odd since "sensor ID" won't be unique per "instance of data", right?

Reply:
In DDS terminology, an instance is a set of samples that all have the same key value. A data sample is an individual message. So you can think of an instance as a group of samples that all have the same key. But this terminology is kind of confusing so let me see if I can make the language better.